src/main/antora/modules/web/pages/servlets/servlets003.adoc

```diff
@@ -1,58 +1,158 @@
 == Sharing Information
 
-Web components, like most objects, usually work with other objects to accomplish their tasks.
-Web components can do so by doing the following.
-
-* Using private helper objects (for example, JavaBeans components).
-
-* Sharing objects that are attributes of a public scope.
-
-* Using a database.
-
-* Invoking other web resources.
-The Jakarta Servlet technology mechanisms that allow a web component to invoke other web resources are described in xref:servlets/servlets.adoc#_invoking_other_web_resources[Invoking Other Web Resources].
-
-=== Using Scope Objects
-
-Collaborating web components share information by means of objects that are maintained as attributes of four scope objects.
-You access these attributes by using the `getAttribute` and `setAttribute` methods of the class representing the scope.
-<<_scope_objects>> lists the scope objects.
-
-[[_scope_objects]]
+Web components, like most objects, usually work with data to accomplish their tasks.
+Web components can store this information in a database or in a scoped bean.
+
+
+=== Using CDI Managed Beans
+
+CDI can be used to define scoped beans which can be injected in any container-managed objects such as web components.
+These scoped beans can then be used to store and transfer information.
+
+CDI defines three basic scopes which can be used for this:
+
+`@jakarta.enterprise.context.RequestScoped`::
+In a servlet container, this is basically mapped to the `jakarta.servlet.ServletRequest`.
+It lives from the time that the client has sent the request until it has retrieved the corresponding response.
+It is not shared elsewhere.
+
+`@jakarta.enterprise.context.SessionScoped`::
+In a servlet container, this is basically mapped to the `jakarta.servlet.http.HttpSession`.
+It lives for as long as the client is interacting with the web application with the same browser instance, and the session hasn't timed out at the server side.
+It is shared among all requests in the same session.
+
+`@jakarta.enterprise.context.ApplicationScoped`::
+In a servlet container, this is basically mapped to the `jakarta.servlet.ServletContext`.
+It lives for as long as the web application lives.
+It is shared among all requests in all sessions.
+
```

Comment:
Are there scenarios where these aren't mapped as described? If not, removing "basically" would make sense.

Reply:
Might be better to remove "basically" here. In a servlet container that's the mapping.

Reply:
Also note that these scopes are just the scopes CDI defines, and which always work for beans injected into a Servlet.

Beans with other (custom) scopes can also be injected, but the developer has to be sure that the scope would indeed be active inside a Servlet.

Reply:
https://github.com/jakartaee/jakartaee-tutorial/pull/36/commits/1319cecc4f88635731a28efadd19ccde8bf6d862